### DISCUSSION TITLE
Support for GeoSPARQL 1.1+

### DISCUSSION THREAD

**Discussion by nicholascar** [dotnetrdf/dotnetrdf#659] [General]
[GeoSPARQL 1.1](https://docs.ogc.org/is/22-047r1/22-047r1.html) was released in early 2024.

It is mostly a fairly simple "tidy-up" update to GeoSPARQL 1.0 (released in 2012) but it includes a number of very useful functions that were just missing from 1.0 and support for widely-used geometry formats like GeoJSON.

dotNetRdf should support 1.1 and watch out for 1.3 that will arrive some time in 1.3.

**Comment by kal** [comment]
Hi @nicholascar 

A couple of initial questions:

1. The current dotNetRDF SPARQL engine does not include any of the GeoSPARQL functions or any functionality for working with geometries. It would seem like implementation of all of this would be quite a large task unless there are any existing .NET libraries that could be leveraged to provide the basic functionality. Are you aware of / able to recommend any such libraries?
2. A quick peruse of the spec (not had time to digest in in any way yet) shows a lot of references to ISO 19125-1:2004, which like most ISO specs is behind a paywall. Is there any openly accessible version of this standard or any openly accessible description of the data structures and functions that it defines?

**Reply by kal** [reply]
And another question that I just thought of after pressing reply:
3. Is there a recommended subset/profile of the standard that we could start with?
